mayastor/src/lvs/lvol.rs

```diff
@@ -398,6 +399,42 @@ impl Lvol {
             )
         };
 
-        info!("Creating snapshot {}", snapshot_name);
+        info!("Creating snapshot {} on {}", snapshot_name, &self);
+    }
+
+    /// Create snapshot for local replica
+    pub async fn create_snapshot_local(
+        &self,
+        io: *mut spdk_sys::spdk_bdev_io,
+        snapshot_name: &str,
+    ) {
+        extern "C" fn snapshot_done_cb(
+            bio_ptr: *mut c_void,
+            _lvol_ptr: *mut spdk_lvol,
+            errno: i32,
+        ) {
+            if errno != 0 {
+                error!("vbdev_lvol_create_snapshot errno {}", errno);
```

Comment:
if there can be an error I guess we should fail the create_snapshot?

Reply:
That's handled later when io_completion_local is called with success = false.